cmd/minishift/cmd/openshift/service.go

```diff
@@ -63,31 +51,23 @@ var serviceCmd = &cobra.Command{
 		api := libmachine.NewClient(constants.Minipath, constants.MakeMiniPath("certs"))
 		defer api.Close()
 
-		url, err := cluster.GetServiceURL(api, namespace, service, serviceURLTemplate)
+		url, err := openshift.GetServiceURL(service, namespace)
 		if err != nil {
 			fmt.Fprintln(os.Stderr, err)
-			if _, ok := err.(cluster.MissingNodePortError); !ok {
-				fmt.Fprintln(os.Stderr, "Verify that Minishift is running and that the correct namespace is specified in the -n option.")
-			}
 			atexit.Exit(1)
 		}
 
-		if https {
-			url = strings.Replace(url, "http", "https", 1)
-		}
 		if serviceURLMode {
 			fmt.Fprintln(os.Stdout, url)
 		} else {
 			fmt.Fprintln(os.Stdout, "Opening the service "+namespace+"/"+service+" in the default browser...")
-			browser.OpenURL(url)
+			browser.OpenURL("http://" + url)
```

Comment:
http hardcoded? routes can be defined as secured...
https://docs.openshift.com/enterprise/3.0/dev_guide/routes.html